status-models/src/main/java/com/telecominfraproject/wlan/status/equipment/report/models/UnmanagedNeighbourEquipmentInfo.java

```diff
@@ -71,21 +72,24 @@ else if (lowestMac != null && lowestMac.getAddressAsLong() > newMac.getAddressAs
 
     @JsonIgnore
     public int getAverageSignalStrenght(RadioType radioType) {
-        int runningTotal = 0;
-        int numEntries = 0;
-
+        List<Integer> rssiList = new ArrayList<>();
+        
         for (NeighbourRadioInfo radioInfo : this.radios) {
             if (radioInfo.getRadioType() == radioType) {
```

Comment:
Also add null checks for radioInfo and radioInfo.getBssIds() here and log/return appropriately if they are null